libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -158,6 +159,90 @@ def map_to_base_relationship(rel: Any) -> Relationship:
     )
 
 
+def _convert_to_graph_document(
+    raw_schema: Dict,
+) -> Tuple[List[Node], Tuple[List[Relationship]]]:
+    # If there are validation errors
+    if not raw_schema["parsed"]:
+        try:
+            argument_json = json.loads(
+                raw_schema["raw"].additional_kwargs["tool_calls"][0]["function"][
+                    "arguments"
+                ]
+            )
+            nodes = []
+            for node in argument_json["nodes"]:
+                if not node.get("id"):  # Id is mandatory
+                    continue
+                nodes.append(
+                    Node(
+                        id=node["id"].title(),
+                        type=node.get("type", "Node").capitalize(),
+                    )
+                )
+            relationships = []
+            for rel in argument_json["relationships"]:
+                # Mandatory props
+                if (
+                    not rel.get("source_node_id")
+                    or not rel.get("target_node_id")
+                    or not rel.get("type")
+                ):
+                    continue
+
+                # Node type copying if needed from node list
+                if not rel.get("source_node_type"):
+                    try:
+                        rel["source_node_type"] = [
+                            el["type"]
+                            for el in argument_json["nodes"]
+                            if el["id"] == rel["source_node_id"]
+                        ][0]
+                    except IndexError:
+                        rel["source_node_type"] = "Node"
```

Comment:
should we also check that the source-node-type is actually the same for the relationship and the node in the nodes list?

Reply:
and what happens if there is not even a node in the list for this rel, should we actually skip the rel or could it be a node that's extracted in a prior part of the document (would it not have to show up in our node list if we have a rel with this start/end-node)?

Reply:
So if there are multiple one nodes with the same id, but different label, we just take the first one.  We don't do any validations, and if the node doesn't show up in the node list, we keep the relationship and use the default node type, which is "Node"... but I'll change it here to None, so that the default type get's used and we don't have the default set in multiple places